go/tools/builders/compilepkg.go

```diff
@@ -343,6 +342,15 @@ func compileArchive(
 		if err != nil {
 			return err
 		}
+		if coverMode != "" {
+			// Compile original source files, not coverage instrumented, for nogo
+			_, goSrcsNogo, _, err = cgo2(goenv, goSrcsNogo, cgoSrcsNogo, cSrcs, cxxSrcs, objcSrcs, objcxxSrcs, sSrcs, hSrcs, packagePath, packageName, cc, cppFlags, cFlags, cxxFlags, objcFlags, objcxxFlags, ldFlags, cgoExportHPath)
+			if err != nil {
+				return err
+			}
+		} else {
+			goSrcsNogo = goSrcs
+		}
```

Comment:
Nit: do we need this else clause? Seems redundant.

```suggestion
		}
```

Reply:
This else clause is necessary, because `goSrcsNogo` was referencing to `goSrcs` before cgo compilation, and here it's updated to `goSrcs` that are cgo compiled, this will ensure cgo enabled targets can run static analysis successfully.